File: src/noc/endpoints/NocNode.cc
#include "noc/endpoints/NocNode.hh"

namespace gem5
{
namespace noc
{

NocNode::NocNode(const Params &p)
    : SimObject(p), clockDomains(p.clockDomains),
      portEndpointNames(p.port_endpoint_names)
{
}

void
NocNode::serialize(CheckpointOut &cp) const
{
    SimObject::serialize(cp);
    serializeNocNodeState(cp);
}

void
NocNode::unserialize(CheckpointIn &cp)
{
    SimObject::unserialize(cp);
    unserializeNocNodeState(cp);
}

void
NocNode::serializeNocNodeState(CheckpointOut &cp) const
{}

void
NocNode::unserializeNocNodeState(CheckpointIn &cp)
{}

} // namespace noc
} // namespace gem5

File: src/noc/endpoints/NocNode.py
from m5.params import *
from m5.SimObject import SimObject


class NocNode(SimObject):
    type = "NocNode"
    cxx_header = "noc/endpoints/NocNode.hh"
    cxx_class = "gem5::noc::NocNode"
    abstract = True

    clockDomains = VectorParam.Int([], "Clock domain(s) in MHz")
    port_endpoint_names = VectorParam.String(
        [], "Endpoint names per port in port order"
    )


